code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -1107,20 +1107,28 @@ func (sc *StorageSmartContract) addChallenge(alloc *StorageAllocation,
 	}
 
 	// remove expired challenges
-	expiredIDs, err := alloc.removeExpiredChallenges(allocChallenges, challenge.Created, challenge.BlobberID)
-	lenExpiredIDs   := len(expiredIDs)
+	expiredIDsMap, totalExpiredCount, err := alloc.removeExpiredChallenges(allocChallenges, challenge.Created, challenge.BlobberID)
 	if err != nil {
 		return common.NewErrorf("add_challenge", "remove expired challenges: %v", err)
 	}
 
+	// maps blobberID to count of its expiredIDs.
+	expiredCountMap := make(map[string]int)
+
 	// TODO: maybe delete them periodically later instead of remove immediately
-	for _, id := range expiredIDs {
+	for bid, IDsPerBlobber := range expiredIDsMap {
+		if _, ok := expiredCountMap[bid]; !ok {
+			expiredCountMap[bid] = len(IDsPerBlobber)
+		}
+		for _, id := range IDsPerBlobber {
 		_, err := balances.DeleteTrieNode(storageChallengeKey(sc.ID, id))
 		if err != nil {
 			if err == util.ErrValueNotPresent {
-				lenExpiredIDs--
+					expiredCountMap[bid]--
+					totalExpiredCount--
```

Comment:
Do not change the expired number as they should be what they are when collected, otherwise the stats will be inconsisted with the stat in allocation and each blobber in MPT. 

If we think there could be duplicate open challenges that could cause the second time of node deleting, then we need to add logs to check and debug to find out why there are duplicate items, not change it here as this will hide the bug if it exists. We can fix it if any in another thread, but the easy way to debug is log the open challenge ID when removing it, and log the ValueNotPresent error, the grafana will persist logs for us so that when the error happens, we can see when was the first time the challenge get deleted.